mp/src/game/shared/momentum/util/engine_patch.cpp

```diff
@@ -0,0 +1,243 @@
+    moduleSize = info.SizeOfImage;
+#else //POSIX
+    if (GetModuleInformation(ENGINE_DLL_NAME, &moduleBase, &moduleSize))
+        return false;
+#endif //WIN32
+
+    return true;
+}
+
+void CEngineBinary::PostInit()
+{
+    CEnginePatch::ApplyAll();
+}
+
+inline bool CEngineBinary::DataCompare(const unsigned char* data, const unsigned char* pattern, const char* mask)
+{
+    for (; *mask != 0; ++data, ++pattern, ++mask)
+        if (*mask == 'x' && *data != *pattern)
+            return false;
+
+    return (*mask == 0);
+}
+
+//---------------------------------------------------------------------------------------------------------
+// Finds a pattern of bytes in the engine memory given a signature and a mask
+// Returns the address of the first (and hopefully only) match with an optional offset, otherwise nullptr
+//---------------------------------------------------------------------------------------------------------
+void* CEngineBinary::FindPattern(const unsigned char* pattern, const char* mask, size_t offset)
+{
+    auto maskLength = strlen(mask);
+    for (size_t i = 0; i <= moduleSize - maskLength; ++i)
+    {
+        auto addr = reinterpret_cast<const unsigned char*>(moduleBase) + i;
+        if (DataCompare(addr, pattern, mask))
+            return const_cast<void*>(reinterpret_cast<const void*>(addr + offset));
+    }
+
+    return nullptr;
+}
+
+CEngineBinary g_EngineBinary;
+
+// Engine Patch format:
+//
+// Patch name
+// Memory signature
+// Signature mask
+// Patch offset
+// Immediate or referenced variable
+// Patch bytes (int/float/char*)
+#ifdef _WIN32
+CEnginePatch g_EnginePatches[] =
+{
+    // Prevent the culling of skyboxes at high FOVs
+    // https://github.com/VSES/SourceEngine2007/blob/master/se2007/engine/gl_warp.cpp#L315
+    // TODO: Use a value derived from FOV instead
+    {
+        "SkyboxCulling",
+        (unsigned char*)"\xF3\x0F\x59\x15\x00\x00\x00\x00\xF3\x0F\x58\xC1\xF3\x0F\x10\x0D",
+        "xxxx????xxxxxxxx",
+        16,
+        false,
+        (float)-1
+    },
+    // Example patch: Trigger "Map has too many brushes" error at 16384 brushes instead of 8192
+    //{
+    //    "BrushLimit",
+    //    (unsigned char*)"\xC2\xEF\x03\x81\xFF\x00\x00\x00\x00",
+    //    "xxxxx????",
+    //    5,
+    //    true,
+    //    16384
+    //},
+    // The same patch as above but using a pure hex value
+    //{
+    //    "BrushLimit2",
+    //    (unsigned char*)"\xC1\xEF\x03\x81\xFF\x00\x00\x00\x00",
+    //    "xxxxx????",
+    //    5,
+    //    true,
+    //    (unsigned char*)"\x00\x40\x00\x00"
+    //}
+};
+#elif __linux__
+CEnginePatch g_EnginePatches[] =
+{
+    // Prevent the culling of skyboxes at high FOVs
+    // https://github.com/VSES/SourceEngine2007/blob/master/se2007/engine/gl_warp.cpp#L315
+    // TODO: Use a value derived from FOV instead
+    {
+        "SkyboxCulling",
+        (unsigned char*)"\xF3\x0F\x59\x0D\x00\x00\x00\x00\xF3\x0F\x58\xC2\xF3\x0F\x58\xC1\xF3\x0F\x10\x0D",
```

Comment:
Does this give a warning/error if the `(unsigned char*)` cast is removed? If it does, then why not just use a regular `char*` parameter. AFAIK there shouldn't be any difference between behavior of `unsigned char*`/`char*` in this context and it would eliminate the ugly cast.

Reply:
It does error out in that case, but as you said using `char*` was not any different in this context.